Source/Core/VideoCommon/VertexLoaderX64.cpp

```diff
@@ -98,30 +97,104 @@ int VertexLoaderX64::ReadVertex(OpArg data, u64 attribute, int format, int count
 
 	int elem_size = 1 << (format / 2);
 	int load_bytes = elem_size * count_in;
-	if (load_bytes > 8)
-		MOVDQU(coords, data);
-	else if (load_bytes > 4)
-		MOVQ_xmm(coords, data);
-	else
-		MOVD_xmm(coords, data);
+	OpArg dest = MDisp(dst_reg, m_dst_ofs);
+
+	native_format->components = count_out;
+	native_format->enable = true;
+	native_format->offset = m_dst_ofs;
+	native_format->type = VAR_FLOAT;
+	native_format->integer = false;
 
-	PSHUFB(coords, M(&shuffle_lut[format][count_in - 1]));
+	m_dst_ofs += sizeof(float) * count_out;
 
-	if (format != FORMAT_FLOAT)
+	if (attribute == DIRECT)
+		m_src_ofs += load_bytes;
+
+	if (cpu_info.bSSSE3)
 	{
-		// Sign extend
+		if (load_bytes > 8)
+			MOVDQU(coords, data);
+		else if (load_bytes > 4)
+			MOVQ_xmm(coords, data);
+		else
+			MOVD_xmm(coords, data);
+
+		PSHUFB(coords, M(&shuffle_lut[format][count_in - 1]));
+
+		// Sign-extend.
 		if (format == FORMAT_BYTE)
 			PSRAD(coords, 24);
 		if (format == FORMAT_SHORT)
 			PSRAD(coords, 16);
+	}
+	else
+	{
+		// SSE2
+		X64Reg temp = XMM1;
+		switch (format)
+		{
+		case FORMAT_UBYTE:
+			MOVD_xmm(coords, data);
+			PXOR(temp, R(temp));
+			PUNPCKLBW(coords, R(temp));
+			PUNPCKLWD(coords, R(temp));
+			break;
+		case FORMAT_BYTE:
+			MOVD_xmm(coords, data);
+			PUNPCKLBW(coords, R(coords));
+			PUNPCKLWD(coords, R(coords));
+			PSRAD(coords, 24);
+			break;
+		case FORMAT_USHORT:
+		case FORMAT_SHORT:
+			switch (count_in)
+			{
+			case 1:
+				LoadAndSwap(32, scratch3, data);
+				MOVD_xmm(coords, R(scratch3));    // ......X.
+				break;
+			case 2:
+				LoadAndSwap(32, scratch3, data);
+				MOVD_xmm(coords, R(scratch3));    // ......XY
+				PSHUFLW(coords, R(coords), 0x24); // ....Y.X.
+				break;
+			case 3:
+				LoadAndSwap(64, scratch3, data);
+				MOVQ_xmm(coords, R(scratch3));    // ....XYZ.
+				MOVDQA(temp, R(coords));          // ......Z.
```

Comment:
Is this necessary?  pshuflw can do a move built-in, so I think you can avoid this, e.g. with:

movq temp, scratch3
pshuflw coords, temp, 0xAC
punpcklqdq coords, temp